### ISSUE TITLE
Missing saveStates

### ISSUE THREAD

**Issue by cullenn** [timebomb0/HordesUIMod#10]
It's really easy to miss calls to `saveState()`. Is it worth revisiting proxies in order to hook one up to call `saveState()` every time state gets set and scrap all other calls to `saveState()`? Feels like it'd be a lot more intuitive to just assume it's being saved every time you set it unless there's a situation where you'd want to set it and not save right away that I'm considering

https://github.com/timebomb0/HordesUIMod/blob/ed4edcbf3c11ddc39c565c0d1ff2188820282e25/src/mods/draggableUI/index.js#L38-L45

**Comment by timebomb0** [comment]
Yeah, it's definitely a potential issue. I originally wanted to avoid this because calls to localStorage.set, or saving state, are a _lot_ more negatively performance impacting than setting an object.

I could run a check every 5 seconds to see if the state has changed, and if it has, save it. Then we wouldn't have to worry about calling state. What do you think?

**Comment by cullenn** [comment]
Oh that makes sense, I wasn't thinking a proxy's `set:` on state would be triggered too often but maybe I'm underestimating that lol, checking every 5 seconds would definitely work as a nice safety net
